src/main/java/wellnus/gamification/util/GamificationStorage.java

```diff
@@ -0,0 +1,54 @@
+package wellnus.gamification.util;
+
+import java.util.ArrayList;
+
+import wellnus.exception.StorageException;
+import wellnus.exception.TokenizerException;
+import wellnus.storage.Storage;
+
+/**
+ * Manages the storage and retrieval of gamification data to and from storage.
+ */
+public class GamificationStorage {
+    private final Storage storage;
+    private final GamificationTokenizer tokenizer;
+
+    /**
+     * Returns an instance of GamificationStorage.
+     * @throws StorageException If Storage class cannot be initialised successfully
+     */
+    public GamificationStorage() throws StorageException {
+        this.storage = new Storage();
+        this.tokenizer = new GamificationTokenizer();
+    }
+
+    /**
+     * Loads tokenized gamification data from storage and detokenizes them back into
+     * a GamificationData object.
+     * @return GamificationData object representing the previously saved gamification statistics
+     * @throws StorageException If data cannot be fetched from storage successfully
+     * @throws TokenizerException If tokenized data cannot be detokenized into a GamificationData object
+     */
+    public GamificationData loadData() throws StorageException, TokenizerException {
+        if (storage.checkFileExists(Storage.FILE_GAMIFICATION)) {
+            ArrayList<String> tokenizedObjects = storage.loadData(Storage.FILE_GAMIFICATION);
+            ArrayList<GamificationData> dataObjects = tokenizer.detokenize(tokenizedObjects);
+            int dataObjectIndex = 0;
```

Comment:
Minor nit: Perhaps you could refactor out this magic literal!

Reply:
oh actually this variable is declared to refactor our this magic literal though? i just kept it within the local scope of the method since it's irrelevant to the rest of the app